extensions-core/lookups-cached-global/src/main/java/org/apache/druid/server/lookup/namespace/JdbcCacheGenerator.java

```diff
@@ -101,23 +109,41 @@
         }
     );
     final String newVersion;
-    if (lastDBUpdate != null) {
-      newVersion = lastDBUpdate.toString();
-    } else {
-      newVersion = StringUtils.format("%d", dbQueryStart);
-    }
-    final CacheScheduler.VersionedCache versionedCache = scheduler.createVersionedCache(entryId, newVersion);
+    CacheScheduler.VersionedCache versionedCache = null;
     try {
-      final Map<String, String> cache = versionedCache.getCache();
-      for (Pair<String, String> pair : pairs) {
-        cache.put(pair.lhs, pair.rhs);
+      if (doIncrementalLoad) {
+        newVersion = StringUtils.format("%d", lastDBUpdate);
+        ConcurrentMap<String, String> newCachedEntries = new ConcurrentHashMap<>();
+        LOG.info(" found %s new incremental entries for %s", pairs.size(), entryId);
+        for (Pair<String, String> pair : pairs) {
+          newCachedEntries.put(pair.lhs, pair.rhs);
+        }
+        versionedCache = entryId.createFromExisitngCache(entryId, newVersion, newCachedEntries);
```

Comment:
Please leave a logging statement similar to `LOG.info("Finished loading %d values for %s", cache.size(), entryId);` in the other branch

Reply:
done